jobs: enable downloading execution detail files

adityamaru opened the pull request:
In https://github.com/cockroachdb/cockroach/pull/106879 we added a table to the `Advanced Debugging`
tab of the job details page. This table lists out all
the execution detail files that are available for the
given job.

This change is a follow up to add download functionality
to each row in the table. The format of the downloaded
file is determined by the prefix of the filename.

A final change to allow users to generate execution details
will be added in the next follow up.

Informs: https://github.com/cockroachdb/cockroach/issues/105076
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107210)
<!-- Reviewable:end -->

adityamaru commented:
https://www.loom.com/share/c89895f44e41462594a22dcd8f72caf0?sid=afc4447c-97e4-4f30-9829-33445aa4f669

adityamaru committed:
jobs: enable downloading execution detail files

In #106879 we added a table to the `Advanced Debugging`
tab of the job details page. This table lists out all
the execution detail files that are available for the
given job.

This change is a follow up to add download functionality
to each row in the table. The format of the downloaded
file is determined by the prefix of the filename.

A final change to allow users to generate execution details
will be added in the next follow up.

Informs: #105076
Release note: None

adityamaru commented:
first commit is from https://github.com/cockroachdb/cockroach/pull/107198 so only second commit needs a review here!

adityamaru commented:
@maryliag based on your offline chat I tried to add a space between `Download` and the icon but I can't seem to figure it out 🤦 . I looked if statement bundles do anything but they seem to also have the same button/icon layout as in the loom.

adityamaru commented:
oh I might have figured it out, pushing something.

adityamaru commented:
here we go!

<img width="1548" alt="Screenshot 2023-07-26 at 4 37 06 PM" src="https://github.com/cockroachdb/cockroach/assets/13837382/1c152877-96e8-4513-95bd-55a077423848">

maryliag reviewed (approved):
Nice! Thanks for also updating name of function to clarify is about `files`.
I will leave to somebody on your team to review the first commit, since I don't know enough to make sure is covering everything, so my review is focused on the ui changes only, which is the second commit

For that I added a few nits, otherwise <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1, 9 of 12 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107210#-:-NaIz-1cC4jjXeoPC20K:bed9dsv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @adityamaru and @dt)

___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx` line 17 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107210#-NaIxJCE54zP9SaZ-0qG:-NaIxJCE54zP9SaZ-0qH:bt6h1a) ([raw file](https://github.com/cockroachdb/cockroach/blob/dee93fcfead89e43dc71a9ce733121e98896f2f3/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx#L17)):*
> ```XML
> import { Button, InlineAlert, Icon } from "@cockroachlabs/ui-components";
> import { Row, Col } from "antd";
> import "antd/lib/icon/style";
> ```

curious, where you getting lint error if you didn't add those? Asking because you just need to import the style when the import was the "antd". The Button and Icon were not, so there is no need
___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.module.scss` line 5 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107210#-NaIyeprDcWkBTKATV5K:-NaIyeprDcWkBTKATV5L:bnviwot) ([raw file](https://github.com/cockroachdb/cockroach/blob/dee93fcfead89e43dc71a9ce733121e98896f2f3/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.module.scss#L5)):*
> ```SCSS
> .crl-job-profiler-view {
>     display: flex;
>     flex-direction: column;
> ```

I don't see you using `crl-job-profiler-view` anywhere, just `crl-job-profiler-view__actions-column`, so these tow lines are not being used anywhere


<!-- Sent from Reviewable.io -->

adityamaru reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107210)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt and @maryliag)

___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx` line 17 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107210#-NaIxJCE54zP9SaZ-0qG:-NaNmU0r1CIcDAD5eZ5a:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dee93fcfead89e43dc71a9ce733121e98896f2f3/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.tsx#L17)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

curious, where you getting lint error if you didn't add those? Asking because you just need to import the style when the import was the "antd". The Button and Icon were not, so there is no need
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.module.scss` line 5 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107210#-NaIyeprDcWkBTKATV5K:-NaNmUgg8MyABJBI1nfK:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dee93fcfead89e43dc71a9ce733121e98896f2f3/pkg/ui/workspaces/cluster-ui/src/jobs/jobDetailsPage/jobProfilerView.module.scss#L5)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

I don't see you using `crl-job-profiler-view` anywhere, just `crl-job-profiler-view__actions-column`, so these tow lines are not being used anywhere
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

adityamaru commented:
Now that this is rebased, I'm going to merge this UI only change.

TFTR!

bors r=maryliag

adityamaru commented:
bors retry
